mongodump/prepare.go

```diff
@@ -193,10 +194,8 @@ func (dump *MongoDump) outputPath(dbName, colName string) string {
 	} else {
 		root = dump.OutputOptions.Out
 	}
-	if dbName == "" {
-		return filepath.Join(root, colName)
-	}
-	return filepath.Join(root, dbName, colName)
+
+	return filepath.Join(root, dbName, util.EscapeCollectionName(colName))
```

Comment:
Is there significance to eliminating the case when `dbName` is empty?

Reply:
I also wonder if we should be percent-encoding the database name.

Reply:
>Is there significance to eliminating the case when dbName is empty?

It's just a bit of cleanup. As per @kmahar 's [comment](https://github.com/mongodb/mongo-tools/pull/126#discussion_r261770219), `filepath.Join` ignores empty strings.

>I also wonder if we should be percent-encoding the database name.

Database names are restricted from having `/\. "$` in them, so any valid db name shouldn't have to be escaped (I think).